modules/F19_cve_aggregator.sh

```diff
@@ -494,15 +491,19 @@ aggregate_versions() {
 
   # sorting and unique our versions array:
   #eval "VERSIONS_CLEANED=($(for i in "${VERSIONS_CLEANED[@]}" ; do echo "\"$i\"" ; done | sort -u))"
-  if [[ -f "$LOG_DIR"/aggregator/versions.tmp ]]; then
+  if [[ -f "$LOG_PATH_MODULE"/versions.tmp ]]; then
     # on old kernels it takes a huge amount of time to query all kernel CVE's. So, we move the kernel entry to the begin of our versions array
-    mapfile -t KERNELS < <(grep kernel "$LOG_DIR"/aggregator/versions.tmp)
-    grep -v kernel "$LOG_DIR"/aggregator/versions.tmp | sort -u > "$LOG_DIR"/aggregator/versions1.tmp
+    mapfile -t KERNELS < <(grep kernel "$LOG_PATH_MODULE"/versions.tmp)
+    grep -v kernel "$LOG_PATH_MODULE"/versions.tmp | sort -u > "$LOG_PATH_MODULE"/versions1.tmp
     for KERNEL in "${KERNELS[@]}"; do
-      sed -i "1s/^/$KERNEL\n/" "$LOG_DIR"/aggregator/versions1.tmp
+      if [[ $( wc -l "$LOG_PATH_MODULE"/versions1.tmp | cut -d" " -f1 ) -eq 0 ]] ; then
```

Comment:
do we really need this? I have not seen any problems with just executing the regex

Reply:
The regex doesn't work, if the version1.tmp is empty.

Reply:
sure ... but then there is no kernel entry and nothing should happen

Reply:
While testing, I run emba with -m s25, which is the kernel module and it found the kernel in the firmware, but then with the old code, the aggregator won't check the kernel in the CVE database. With the new code, it works.